mlrun/datastore/sources.py

```diff
@@ -826,6 +875,252 @@ def add_nuclio_trigger(self, function):
+                              db_name="db_name",
+                              collection_name='collection',
+                              create_collection=True,
+                              override_collection=True)
+
+        :param name:                source name
+        :param query:               dictionary query for mongodb
+        :param chunksize:           number of rows per chunk (default large single chunk)
+        :param key_field:           the column to be used as the key for events. Can be a list of keys.
+        :param time_field:          the column to be parsed as the timestamp for events. Defaults to None
+        :param start_time:          filters out data before this time
+        :param end_time:            filters out data after this time
+        :param schedule:            string to configure scheduling of the ingestion job. For example '*/30 * * * *' will
+                                    cause the job to run every 30 minutes
+        :param connection_string:   string connection to mongodb database.
+                                    If not set, the MONGODB_CONNECTION_STR environment variable will be used.
+        :param db_name:             the name of the database to access
+        :param collection_name:     the name of the collection to access, from the current database
+        """
+
+        connection_string = connection_string or os.getenv(
+            self._MONGO_CONNECTION_STRING_ENV_VAR
+        )
+        if connection_string is None:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"cannot specify without connection_string arg or secret {self._MONGO_CONNECTION_STRING_ENV_VAR}"
+            )
+        attrs = {
+            "query": query,
+            "collection_name": collection_name,
+            "db_name": db_name,
+            "max_results": max_results_for_table,
+            "chunksize": chunksize,
+            "spark_options": spark_options,
+            "connection_string": connection_string,
+        }
+        attrs = {key: value for key, value in attrs.items() if value is not None}
+        super().__init__(
+            name,
+            attributes=attrs,
+            key_field=key_field,
+            time_field=time_field,
+            schedule=schedule,
+            start_time=start_time,
+            end_time=end_time,
+        )
+
+    def to_dataframe(self) -> Union[pd.DataFrame, _MondoDBIterator]:
+        from pymongo import MongoClient
+
+        query = self.attributes.get("query")
+        db_name = self.attributes.get("db_name")
+        collection_name = self.attributes.get("collection_name")
+        connection_string = self.attributes.get("connection_string")
+        chunksize = self.attributes.get("chunksize")
+
+        if collection_name and db_name:
+            # connect and read the desired collection
+            mongodb_client = MongoClient(connection_string)
```

Comment:
that's how mongo suggest to work with the client? each time create a client and close it ASAP. I mean in some situations maybe it makes more sense to initialize a client, keep it and close it once you are done after multiple calls?